src/moonlink/src/storage/mooncake_table.rs

```diff
@@ -843,14 +940,20 @@ impl MooncakeTable {
 
         // Re-initialize mooncake table fields.
         self.next_snapshot_task = SnapshotTask::new(self.metadata.config.clone());
+        // Carry forward the commit baseline
+        // This is important if we have a pending flush that will be added to the next snapshot task.
+        self.next_snapshot_task.new_commit_lsn = next_snapshot_task.new_commit_lsn;
```

Comment:
- You need to rename the variable, it's not only **new** LSNs any more
- Could you please elaborate what if we don't carry over? 

TBH
> This is important if we have a pending flush that will be added to the next snapshot task.

Tells me nothing